openmmforcefields/generators/template_generators.py

```diff
@@ -1030,12 +1030,23 @@ def as_attrib(quantity):
                 element=element_symbol, mass=as_attrib(atom.mass))
             atom_type.set('class', atom.typename) # 'class' is a reserved Python keyword, so use alternative API
 
+        supported_forces = {
+            "NonbondedForce",
+            "HarmonicAngleForce",
+            "HarmonicBondForce",
+            "PeriodicTorsionForce",
+        }
+
         # Compile forces into a dict
         forces = dict()
         for force in system.getForces():
             force_name = force.__class__.__name__
+
             if force_name in forces:
-                raise Exception("Two instances of force {force_name} appear in System")
+                raise Exception(f"Two instances of force {force_name} appear in System")
+            if force_name not in supported_forces:
+                raise Exception(f"Custom forces not supported. Found force of type {force_name}.")
```

Comment:
I should probably add I don't think this is the right exception to raise, I'm just not sure which it should be

Reply:
Eh, this package doesn't define many exceptions, but I made a new one